src/main/java/com/arpnetworking/metrics/mad/Aggregator.java

```diff
@@ -279,12 +228,182 @@ private Aggregator(final Builder builder) {
     private final ImmutableMap<Pattern, ImmutableSet<Statistic>> _statistics;
     private final LoadingCache<String, Optional<ImmutableSet<Statistic>>> _cachedSpecifiedStatistics;
     private final LoadingCache<String, Optional<ImmutableSet<Statistic>>> _cachedDependentStatistics;
-    private final Map<Key, List<ActorRef>> _periodWorkerActors = Maps.newConcurrentMap();
 
     private static final StatisticFactory STATISTIC_FACTORY = new StatisticFactory();
-    private static final Timeout SHUTDOWN_TIMEOUT = Timeout.apply(1, TimeUnit.SECONDS);
+    private static final Duration SHUTDOWN_TIMEOUT = Duration.ofSeconds(30);
     private static final Logger LOGGER = LoggerFactory.getLogger(Aggregator.class);
 
+    private static final class ShutdownAggregator implements Serializable {
```

Comment:
Why is this message type a singleton instead of a string?

I noticed that string types are used for singleton messages in the other actor.

Reply:
Generally, string instances are okay for internal messages. I had this originally also being sent from Aggregator to PeriodWorker (just like it does in CAGG). However, it was changed to just a PoisonPill. I suspect in the future we will be sending ShutdownAggregator to the PeriodWorker to allow it to kill itself gracefully (including flushing of state to disk).